lib/codeowners.js

```diff
@@ -1,65 +1,89 @@
 const minimatch = require('minimatch');
-
-const EMAIL_FORMAT = /^\S+@\S+$/;
+const yaml = require('js-yaml');
 
 module.exports = function(contents) {
-  const globOwners = new Map();
+  const globData = new Map();
 
-  contents.split('\n').forEach((line) => {
-    const trimmedLine = line.trim();
+  const pathsForLabels = yaml.safeLoad(Buffer.from(contents).toString());
 
-    // discard comments
-    if (trimmedLine.startsWith('#')) {
-      return;
+  for (const label in pathsForLabels) {
+    if (!pathsForLabels.hasOwnProperty(label)) {
+      continue;
     }
 
-    const lineGlobs = [];
-    const lineOwners = [];
-
-    // split by any number of whitespaces
-    const tokens = trimmedLine.split(/(\s+)/);
-    tokens.forEach((token) => {
-      const isEmail = EMAIL_FORMAT.exec(token);
-      if (isEmail) {
-        // email
-        lineOwners.push(token);
-      } else if (token.startsWith('@')) {
-        // github user or team
-        lineOwners.push(token);
-      } else if (token !== null && token.trim() !== '') {
-        // glob pattern
-        lineGlobs.push(token);
-      }
-    });
+    const pathsForLabel = pathsForLabels[label];
 
-    lineGlobs.forEach((glob) => {
-      globOwners[glob] = lineOwners;
-    });
-  });
+    const labels = Object.keys(pathsForLabel);
```

Comment:
pathsForLabel is a list, isn't it? I've missed something

```
- "label"
  - "path1"
  - "path2"
  - "path3"
  - "path4"
```

Reply:
Loving NodeJS????